server/src/resolvers/getVariantsResolver/adapters/g4rdAdapter.ts

```diff
@@ -92,17 +95,52 @@ const getG4rdNodeQuery = async ({
 
     // Get patients info
     if (G4RDVariantQueryResponse) {
-      const individualIds = G4RDVariantQueryResponse.data.results.map(
-        v => v.individual.individualId
-      );
-      const patientUrl = `${process.env.G4RD_URL}/rest/patients/fetch?${individualIds
-        .map(id => `id=${id}`)
-        .join('&')}`;
-
-      G4RDPatientQueryResponse = await axios.get<G4RDPatientQueryResult>(patientUrl, {
-        headers: { Authorization, 'Content-Type': 'application/json', Accept: 'application/json' },
-        httpsAgent: new https.Agent({ rejectUnauthorized: false }),
-      });
+      // Get all unique individual Ids.
+      const individualIds = G4RDVariantQueryResponse.data.results.reduce(function (prev, curr) {
+        if (curr.individual.individualId && prev.indexOf(curr.individual.individualId) === -1)
+          return [...prev, curr.individual.individualId];
+        else return prev;
+      }, [] as Array<string>);
+
+      if (individualIds.length > 0) {
+        const patientUrl = `${process.env.G4RD_URL}/rest/patients/fetch?${individualIds
+          .map(id => `id=${id}`)
+          .join('&')}`;
+
+        G4RDPatientQueryResponse = await axios.get<G4RDPatientQueryResult>(patientUrl, {
+          headers: {
+            Authorization,
+            'Content-Type': 'application/json',
+            Accept: 'application/json',
+          },
+          httpsAgent: new https.Agent({ rejectUnauthorized: false }),
```

Comment:
Do they still have certificate issues?

Reply:
Good point. I don't remember clearly the conversion around the certificate issue. Maybe it was mentioned before I joined the project. I just removed `httpsAgent` because the queries to the PT endpoint work fine without it.